app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,37 @@
+class PostsController < ApplicationController
+  expose_decorated :post
+
+  before_action :authorize_resource, only: %i[new edit create update]
```

Comment:
It would be better to add authorization to all actions in ApplicationController and skip it only for some actions.
```suggestion
  skip_before_action :authorize_resource, only: :show
```